modules/mbedtls/Kconfig

```diff
@@ -202,6 +202,13 @@ config MBEDTLS_HEAP_SIZE
 	  be needed. For some dedicated and specific usage of mbedtls API, the
 	  1000 bytes might be ok.
 
+config MBEDTLS_INIT
+	bool "Initialize mbed TLS automatically at boot"
+	default y
+	help
```

Comment:
I believe this should be dependent upon `MBEDTLS_ENABLE_HEAP`

If there is no heap enabling in Zephyr mbedTLS, then this call is empty:
https://github.com/zephyrproject-rtos/zephyr/blob/4da89756b3b5853ec4773a20eec814c7af9062a4/modules/mbedtls/zephyr_init.c#L88
as can be seen here:
https://github.com/zephyrproject-rtos/zephyr/blob/4da89756b3b5853ec4773a20eec814c7af9062a4/modules/mbedtls/zephyr_init.c#L44-L46

Ref:
https://github.com/zephyrproject-rtos/zephyr/blob/79fccbb9426385c80d5c1001a3f612eeff4cd625/modules/mbedtls/Kconfig#L180-L181

Please reason if you disagree that `MBEDTLS_INIT` should not have this dependency.

Reply:
_mbedtls_init() does more than just init the heap, it can now also call mbedtls_debug_set_threshold(). Therefore, _mbedtls_init() needs to be called whether we are using the heap or not. So MBEDTLS_INIT doesn't need to be dependant on MBEDTLS_ENABLE_HEAP.